src/transaction.rs

```diff
@@ -1,27 +1,93 @@
-//pub mod slip {
-//}
+use serde::{Serialize, Deserialize};
+use crate::hop::{Hop};
+use crate::slip::{Slip};
+use crate::helper::{create_timestamp};
+use crate::crypto::{Signature, PublicKey};
 
-#[derive(Debug)]
-pub struct Transaction {
-    
+#[derive(Serialize, Deserialize, PartialEq, Debug, Copy, Clone)]
+pub enum TransactionBroadcastType {
+  Normal,
+  GoldenTicket,
+  Fee,
+  Rebroadcast,
+  VIP,
+  GoldenChunk,
 }
-impl Transaction {
-    
-//     new(type TxTypeEnum, sig String, slips [Slip])
-// deserialize(Stream or String)
-// serialize() -> Stream or String
-// validate()
-// getType()
-// addSlip(Slip)
-// getSlips -> [Slip]
-// pushToPath(address String)
-// createRebroadcastTxFrom() -> Transaction
-// Notable missing things: from, to, timestamp
 
+#[derive(Serialize, Deserialize, PartialEq, Debug)]
+pub struct TransactionBody {
+    id:   u32,
+    ts:   u64,
+    pub   to: Vec<Slip>,
+    pub   from: Vec<Slip>,
+    sig:  Signature,
+    ver:  f32,
+    pub typ:  TransactionBroadcastType,
+    path: Vec<Hop>,
```

Comment:
I honestly have no idea about path and where it should go. This is something I need to understand...